packages/nimble-components/src/table-column/enum-text/cell-view/index.ts

```diff
@@ -0,0 +1,52 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { styles } from '../../text-base/cell-view/styles';
+import { template } from '../../text-base/cell-view/template';
+import type {
+    TableColumnEnumCellRecord,
+    TableColumnEnumColumnConfig
+} from '../../enum-base';
+import { TableColumnTextCellViewBase } from '../../text-base/cell-view';
+import type { MappingTextConfig } from '../../enum-base/models/mapping-text-config';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-enum-text-cell-view': TableColumnEnumTextCellView;
+    }
+}
+
+/**
+ * A cell view for displaying mapped text
+ */
+export class TableColumnEnumTextCellView extends TableColumnTextCellViewBase<
+TableColumnEnumCellRecord,
+TableColumnEnumColumnConfig
+> {
+    private columnConfigChanged(): void {
+        this.updateText();
+    }
+
+    private cellRecordChanged(): void {
+        this.updateText();
+    }
+
+    private updateText(): void {
+        const value = this.cellRecord.value;
+        if (value === undefined || value === null) {
+            this.text = '';
+            return;
+        }
+
+        const config = this.columnConfig?.mappingConfigs.get(value);
+        this.text = config ? (config as MappingTextConfig).text : '';
```

Comment:
Treat the element more standalone and behave robustly. So an instanceof check instead of a cast / assertion

Reply:
Could you explain what you mean by "treat the element more standalone?" I understand what change you're suggesting, but not the motivation.

Reply:
The cast for `as MappingTextConfig` is knowledge outside the component. The property accepts any type of MappingConfig based on it's type. Based solely on the component on it's own and it's types it can't assume that only MappingTextConfig are assigned.